This diff introduces a subtle but important shift in how WordPress ha…

pbearne opened the pull request:
…ndles dates and times associated with posts.  Instead of referring to the "date the post was written," the functions and filter descriptions now refer to the "date of the post."  This change likely reflects a move towards greater flexibility in how dates are used, accommodating scenarios where the displayed date might not strictly correspond to the writing date (e.g., scheduled posts, backdated posts, or content where the "date" represents something other than creation).

Here's a breakdown of the changes:

* **Function and Filter Descriptions:** Docblocks for `the_date()`, `get_the_date()`, `the_time()`, `get_the_time()`, `get_post_time()`, `the_weekday()`, and `the_weekday_date()` have been updated to reflect the new "date of the post" language.  The filters associated with these functions have also received corresponding updates.  This change is primarily semantic.

* **`get_post_time()` Docblock Update:**  The description for `get_post_time()` is also updated, clarifying that the `$format` parameter accepts standard PHP date formats or 'G' (Unix timestamp) or 'U' (seconds since the Unix Epoch).  The wording emphasizes that this function retrieves the *localized* date of the post.

The changes don't introduce new functionality or modify existing behavior.  They clarify the purpose of these functions and make the documentation more accurate and flexible.  The shift to "date of the post" acknowledges that the displayed date might have different meanings depending on the context.

Trac ticket: https://core.trac.wordpress.org/ticket/51289

Commit:
This diff introduces a subtle but important shift in how WordPress handles dates and times associated with posts.  Instead of referring to the "date the post was written," the functions and filter descriptions now refer to the "date of the post."  This change likely reflects a move towards greater flexibility in how dates are used, accommodating scenarios where the displayed date might not strictly correspond to the writing date (e.g., scheduled posts, backdated posts, or content where the "date" represents something other than creation).

Here's a breakdown of the changes:

* **Function and Filter Descriptions:** Docblocks for `the_date()`, `get_the_date()`, `the_time()`, `get_the_time()`, `get_post_time()`, `the_weekday()`, and `the_weekday_date()` have been updated to reflect the new "date of the post" language.  The filters associated with these functions have also received corresponding updates.  This change is primarily semantic.

* **`get_post_time()` Docblock Update:**  The description for `get_post_time()` is also updated, clarifying that the `$format` parameter accepts standard PHP date formats or 'G' (Unix timestamp) or 'U' (seconds since the Unix Epoch).  The wording emphasizes that this function retrieves the *localized* date of the post.

The changes don't introduce new functionality or modify existing behavior.  They clarify the purpose of these functions and make the documentation more accurate and flexible.  The shift to "date of the post" acknowledges that the displayed date might have different meanings depending on the context.

audrasjb commented on src/wp-includes/general-template.php:
```diff
@@ -2517,7 +2517,7 @@ function the_date_xml() {
 
 /**
- * Displays or retrieves the date the current post was written (once per date)
+ * Displays or retrieves the date of the post (once per date)
```
Missing period ;)

audrasjb commented on src/wp-includes/general-template.php:
```diff
@@ -2730,12 +2730,12 @@ function get_the_time( $format = '', $post = null ) {
  *
- * @param string      $format    Optional. Format to use for retrieving the time the post
- *                               was written. Accepts 'G', 'U', or PHP date format. Default 'U'.
+ * @param string      $format    Optional. Format to use for retrieving the time the post.
```
the time **of** the post ?

audrasjb commented on src/wp-includes/general-template.php:
```diff
@@ -3004,19 +3004,20 @@ function the_weekday() {
  * @param string $before Optional. Output before the date. Default empty.
  * @param string $after  Optional. Output after the date. Default empty.
+ *
+ *@global string    $previousweekday The day of the previous post in the loop.
```
needs a space between * and @ ;)

audrasjb reviewed (changes requested):
Some small changes and you should be good to go

Commit:
This diff for `wp-includes/general-template.php` shows minor documentation improvements for WordPress functions related to date and time display:

* **`the_date()`**: The docblock description is slightly improved for clarity, changing "Displays or retrieves the date of the post (once per date)" to "Displays or retrieves the date of the post (once per date)."  The practical effect of this change is negligible, as it's just adding a period.

* **`get_the_date()`**: The `$format` parameter description is updated for accuracy. It changes from "Format to use for retrieving the time the post" to "Format to use for retrieving the *localized* date of the post."  This clarifies that the returned date is adjusted for the blog's timezone settings, not necessarily GMT.  This is a helpful clarification for developers.

* **`the_modified_date()`**: A missing asterisk before `@global` in the docblock is added for consistency. This is purely a stylistic fix and has no functional impact.

In summary, these changes enhance the clarity and accuracy of the documentation but don't introduce any functional code modifications.  They are helpful for developers understanding the nuances of date and time handling within WordPress templates.

audrasjb reviewed (approved):
Nice!

audrasjb commented:
committed in https://core.trac.wordpress.org/changeset/59691
